### PR TITLE
[FAB-22] User auth

### PR DESCRIPTION
**Pull request by daystram**

### CODE REVIEW

File: fableous-fe/src/Api.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ const baseWS =
     ? `${process.env.REACT_APP_BACKENDWS}`
     : `wss://${window.location.hostname}`;
 
-const apiClient = axios.create({
+export const apiClient = axios.create({
```
Comment (daystram) [inline fableous-fe/src/Api.tsx:14]: Don't need to export this?

File: fableous-fe/src/containers/LoginPage.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+});
+
+export default function LoginPage() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const postLogin = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    apiClient
+      .post("/api/auth/login", {
```
Comment (daystram) [inline fableous-fe/src/containers/LoginPage.tsx:36]: Use restAPI.postLogin from Api.tsx

File: fableous-fe/src/containers/RegisterPage.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+export default function RegisterPage() {
+  const [userName, setUsername] = useState("");
+  const [eMail, setEmail] = useState("");
+  const [passWord, setPassword] = useState("");
+
+  const postRegister = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    apiClient
+      .post("/api/auth/register", {
```
Comment (daystram) [inline fableous-fe/src/containers/RegisterPage.tsx:37]: Use restAPI.postRegister from Api.tsx

File: fableous-fe/src/Api.tsx (hunk 1 above)
Comment (wahyuadt) [reply fableous-fe/src/Api.tsx:14]: 👽

File: fableous-fe/src/containers/RegisterPage.tsx (hunk 3 above)
Comment (nicolauscg) [reply fableous-fe/src/containers/RegisterPage.tsx:37]: replace string with restApi.auth.postRegister().url

File: fableous-fe/src/Api.tsx (hunk 1 above)
Comment (nicolauscg) [reply fableous-fe/src/Api.tsx:14]: yup my bad, should add axios.defaults.baseURL = baseAPI;

remove the export and apiClient could just be axios.create()

File: fableous-fe/src/containers/RegisterPage.tsx (hunk 3 above)
Comment (daystram) [reply fableous-fe/src/containers/RegisterPage.tsx:37]: Like this instead?
``` ts
axios.request(restAPI.auth.postRegister({ ... }))
```

**Review by wahyuadt — APPROVED** [review]

**Comment by nicolauscg** [comment]
I tried running on localhost, do we want to implement the following?
- redirect to homepage after successful login
- if login page is visited and user is already logged in, redirect to homepage too
- add login and register button to navbar only if user not logged in
- logout button on navbar if user logged in, which just clears token in localstorage

**Comment by nicolauscg** [comment]
I also tried to do a post req in homepage and did not find the token in header of request, is this in scope?

```
  const testPost = () => {
    axios.post("/api", {});
  };

  useEffect(() => {
    testPost();
  }, []);
```
mayba need interceptor along the lines of below? (copied from impensa)

```
axios.interceptors.request.use(
  config => {
    if (isLoggedIn()) {
      const userObject = getUserObject();
      config.headers.Authorization = `Bearer ${userObject.token}`;
    } else {
      clearUserObject();
    }

    return config;
  },
  error => Promise.reject(error)
); 
```

File: fableous-fe/src/Auth.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { AxiosResponse } from "axios";
+
+const TOKEN_KEY = "token";
+
+const auth = {
+  saveToken: (token: AxiosResponse<any>) => {
+    localStorage.setItem(TOKEN_KEY, token.headers.authorization);
```
Comment (nicolauscg) [inline fableous-fe/src/Auth.tsx:7]: one more thing @wahyuadt , in this file u save the token "Bearer abcdef..." as is, that exact format (with Bearer) is needed for rest API requests, but for websocket requests, the only thing needed it the token part (WITHOUT Bearer), so I think its better to save only the "abcdef..." part, and in interceptor, append it to "Bearere "

**Pull request merged**